dolphin/api/schemas/storages.py

```diff
@@ -36,3 +36,23 @@
     'required': ['host', 'port', 'username', 'password', 'vendor', 'model'],
     'additionalProperties': False
 }
+
+update = {
+    'type': 'object',
+    'properties': {
+        'host': parameter_types.hostname_or_ip_address,
+        'port': parameter_types.tcp_udp_port,
+        'username': {'type': 'string', 'minLength': 1, 'maxLength': 255},
+        'password': {'type': 'string'},
+        'extra_attributes': {
+            'type': 'object',
+            'patternProperties': {
+                '^[a-zA-Z0-9-_:. ]{1,255}$': {
+                    'type': 'string', 'maxLength': 255
+                }
+            }
+        }
+    },
+    'required': ['username', 'password'],
```

Comment:
how do you think if user deesn't want to update name or password. For example, admin move the device and change the port or ip.

Reply:
The intend is to make sure who ever updates access_info should have the credential of staorage.. If we do not mandate credential to update any one without knowing credentail also can update access_info .. What do you think?